astropy/units/quantity.py

```diff
@@ -1501,7 +1501,6 @@ def _wrap_function(self, function, *args, **kwargs):
             Result of the function call, with the unit set properly.
         """
         unit = kwargs.pop('unit', self.unit)
```

Comment:
I'd move `unit` out as well, and do an `if unit is None: unit = self.unit`

Reply:
I wasn't sure if `None` isn't a valid input for unit here. But if it isn't I can change it :)

Reply:
No, it cannot be `None`, which is why I suggested
```
if unit is None:
    unit = self.unit
```
(bit like you do for `new_format` in `Time`)